pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -369,88 +457,114 @@ func (r *ReconcileIBMBlockCSI) reconcileServiceAccount(instance *ibmblockcsi.IBM
-			//recLogger.Info("Skip reconcile: ClusterRole already exists", "Name", cr.GetName())
+			//logger.Info("Skip reconcile: ClusterRole already exists", "Name", cr.GetName())
 		}
 	}
 
 	return nil
 }
 
-func (r *ReconcileIBMBlockCSI) reconcileClusterRoleBinding(instance *ibmblockcsi.IBMBlockCSI) error {
-	recLogger := log.WithValues("Resource Type", "ClusterRoleBinding")
+func (r *ReconcileIBMBlockCSI) deleteClusterRolesAndBindings(instance *ibmblockcsi.IBMBlockCSI) error {
+	if err := r.deleteClusterRoleBindings(instance); err != nil {
+		return err
+	}
 
-	externalProvisioner := instance.GenerateExternalProvisionerClusterRoleBinding()
-	externalAttacher := instance.GenerateExternalAttacherClusterRoleBinding()
-	externalSnapshotter := instance.GenerateExternalSnapshotterClusterRoleBinding()
-	externalResizer := instance.GenerateExternalResizerClusterRoleBinding()
-	controllerSCC := instance.GenerateSCCForControllerClusterRoleBinding()
-	nodeSCC := instance.GenerateSCCForNodeClusterRoleBinding()
+	if err := r.deleteClusterRoles(instance); err != nil {
+		return err
+	}
+	return nil
+}
 
-	for _, crb := range []*rbacv1.ClusterRoleBinding{
+func (r *ReconcileIBMBlockCSI) deleteClusterRoles(instance *ibmblockcsi.IBMBlockCSI) error {
+	logger := log.WithName("deleteClusterRoles")
+
+	clusterRoles := r.getClusterRoles(instance)
+
+	for _, cr := range clusterRoles {
+		found := &rbacv1.ClusterRole{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		}, found)
+		if err != nil && errors.IsNotFound(err) {
+			continue
+		} else if err != nil {
+			logger.Error(err, "failed to get ClusterRole", "Name", cr.GetName())
+			return err
+		} else {
+			logger.Info("deleting ClusterRole", "Name", cr.GetName())
+			if err := r.client.Delete(context.TODO(), found); err != nil {
+				logger.Error(err, "failed to delete ClusterRole", "Name", cr.GetName())
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileIBMBlockCSI) getClusterRoles(instance *ibmblockcsi.IBMBlockCSI) []*rbacv1.ClusterRole {
+	externalProvisioner := instance.GenerateExternalProvisionerClusterRole()
+	externalAttacher := instance.GenerateExternalAttacherClusterRole()
+	externalSnapshotter := instance.GenerateExternalSnapshotterClusterRole()
+	externalResizer := instance.GenerateExternalResizerClusterRole()
+	controllerSCC := instance.GenerateSCCForControllerClusterRole()
+	nodeSCC := instance.GenerateSCCForNodeClusterRole()
+
+	return []*rbacv1.ClusterRole{
 		externalProvisioner,
 		externalAttacher,
 		externalSnapshotter,
 		externalResizer,
 		controllerSCC,
 		nodeSCC,
-	} {
-		if err := controllerutil.SetControllerReference(instance.Unwrap(), crb, r.scheme); err != nil {
-			return err
-		}
+	}
+}
+
+func (r *ReconcileIBMBlockCSI) reconcileClusterRoleBinding(instance *ibmblockcsi.IBMBlockCSI) error {
+	recLogger := log.WithValues("Resource Type", "ClusterRoleBinding")
```

Comment:
What does the "rec" mean in `recLogger`? Short for reconcile? recursive? recording?
I suggest removing it as in other logger variables or writing the full word.

Reply:
removed